### PR TITLE
Optimize `nix build` times

### PR DESCRIPTION
**Pull request by jimvdl**
@PgBiel and I fixed the long build times by using [`crane`](https://github.com/ipetkov/crane), this significantly improved caching. The total runtime of the CI should roughly be 2 minutes.

Note: the first run will take the longest because it has to rebuild entirely from scratch and repopulate the cache, it will take a couple of passes until the cache has all of the dependencies, any subsequent runs should be completed in under a minute.

Fixes: #2047

### CODE REVIEW

**Review by alerque — COMMENTED** [review]
Having not actually given this a test drive, looking over the code it looks sane to me.

I believe this PR can be marked as fixing #2047.

File: flake.nix [hunk 1]
```diff
@@ -1,103 +1,116 @@
...
+        overlays.default = _: prev: {
+          typst-dev = packageFor prev;
         };
-      });
+      };
 
-      formatter = eachSystem (pkgs: pkgs.nixpkgs-fmt);
+      perSystem = { pkgs, ... }:
+        let
+		  inherit (pkgs) lib;
```
Comment (PgBiel) [inline flake.nix:93]: quick indentation fix :)
```suggestion
          inherit (pkgs) lib;
```

File: flake.nix [hunk 2]
```diff
@@ -1,103 +1,116 @@
...
-      });
+            buildInputs = lib.optionals pkgs.stdenv.isDarwin [
+              pkgs.darwin.apple_sdk.frameworks.CoreServices
+              pkgs.libiconv
+            ];
+          };
+        };
     };
-}
+}
```
Comment (PgBiel) [inline flake.nix:116]: missing final newline

**Review by PgBiel — COMMENTED** [review]
:+1:

**Comment by laurmaedje** [comment]
Thanks to you two for investigating. :)

**Pull request merged**